spec/controllers/observations_controller_spec.rb

```diff
@@ -43,87 +43,141 @@
+    @request.env["devise.mapping"] = Devise.mappings[:user]
+  end
+
   describe "GET #index" do
-    it "assigns all observations as @observations" do
-      @request.env["devise.mapping"] = Devise.mappings[:user]
-      sign_in admin
-      observation = Observation.create! valid_attributes
-          get :index, params: {}, session: valid_session
-          expect(assigns(:observations)).to eq(Observation.all)
+    context "researcher account" do
+      it "assigns all observations as @observations" do
+        sign_in researcher
+        Observation.create! valid_attributes
+        get :index, params: {}, session: valid_session
+        expect(assigns(:observations)).to eq(Observation.all)
+      end
+    end
+
+    context "scientist account" do
+      it "denies access" do
+        sign_in scientist
+        get :index
+        expect(response).to redirect_to(root_url)
+      end
     end
   end
 
   describe "GET #show" do
-    it "assigns the requested observation as @observation" do
-      observation = Observation.create! valid_attributes
-          get :show, params: {id: observation.to_param}, session: valid_session
-          expect(assigns(:observation)).to eq(observation)
+    context "researcher account" do
+      it "assigns the requested observation as @observation" do
+        sign_in researcher
+        observation = Observation.create! valid_attributes
+        get :show, params: {id: observation.to_param}, session: valid_session
+        expect(assigns(:observation)).to eq(observation)
+      end
+    end
+    context "scientist account" do
+      it "denies access" do
+        sign_in scientist
+        observation = Observation.create! valid_attributes
+        get :show, params: {id: observation.to_param}
+        expect(response).to redirect_to(root_url)
+      end
     end
   end
 
   describe "GET #new" do
-    it "assigns a new observation as @observation" do
-          get :new, params: {}, session: valid_session
-          expect(assigns(:observation)).to be_a_new(Observation)
+    context "researcher account" do
+      it "assigns a new observation as @observation" do
+        sign_in researcher
+        get :new
+        expect(response).to render_template(:new)
+        expect(assigns(:observation)).to be_a_new(Observation)
+      end
+    end
+    context "scientist account" do
+      it "allows access" do
+        sign_in scientist
+        get :new
+        expect(assigns(:observation)).to be_a_new(Observation)
+        expect(response).to render_template(:new)
+      end
+    end
+    context "no account" do
+      it "allows access" do
+        get :new
+        expect(response).not_to render_template(:new)
+        expect(response).to redirect_to(root_url)
+        
+        # TODO: figure out why the observation is still set, even when
+        # the user does not have permission. By my reasoning, this
+        # assertion should FAIL!
+        expect(assigns(:observation)).to be_a_new(Observation)
```

Comment:
Why is a new observation successfully assigned, even though the template redirects to root? I couldn't find any of the methods in the controller being called, but commenting out the entire class results in this assertion failing.